shopinvader_api_payment_cart/routers/cart.py

```diff
@@ -0,0 +1,55 @@
+# Copyright 2024 ACSONE SA (https://acsone.eu).
+# @author Stéphane Bidoul <stephane.bidoul@acsone.eu>
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from typing import Annotated
+
+from fastapi import Depends
+
+from odoo import api
+
+from odoo.addons.base.models.res_partner import Partner as ResPartner
+from odoo.addons.fastapi.dependencies import (
+    authenticated_partner,
+    authenticated_partner_env,
+)
+from odoo.addons.payment import utils as payment_utils
+from odoo.addons.shopinvader_api_cart.routers import cart_router
+from odoo.addons.shopinvader_api_payment.routers.utils import Payable
+from odoo.addons.shopinvader_api_payment.schemas import PaymentData
+
+
+@cart_router.get("/current/payable")
+@cart_router.get("/{uuid}/payable")
+def init(
+    env: Annotated[api.Environment, Depends(authenticated_partner_env)],
+    partner: Annotated["ResPartner", Depends(authenticated_partner)],
+    uuid: str | None = None,
+) -> PaymentData | None:
+    """Prepare payment data for the current cart.
+
+    This route is authenticated, so we can verify the cart
+    is accessible by the authenticated partner.
+    """
+    cart = env["sale.order"]._find_open_cart(partner.id, uuid)
+    if not cart:
+        return None
```

Comment:
```suggestion
        raise HTTPException(status_code=404)?
```

Because trying to pay a missing or empty cart is an error?